### PR TITLE
Bake backend config into ServicePort

### PR DESCRIPTION
**Pull request by MrHohn**
Thinking it might be great to separate this out of the actual backend service feature implementation. @rramkumar1 Feel free to close this if you have code changes ready, I remembered you were doing something similar...
@bowei

### CODE REVIEW

File: pkg/controller/translator/translator.go [hunk 1]
```diff
@@ -46,25 +48,28 @@ func NewTranslator(
...
-	podLister      cache.Indexer
-	endpointLister cache.Indexer
-	negEnabled     bool
+	namer               *utils.Namer
+	svcLister           cache.Indexer
+	nodeLister          cache.Indexer
+	podLister           cache.Indexer
+	endpointLister      cache.Indexer
+	backendConfigLister cache.Indexer
+	negEnabled          bool
```
Comment (rramkumar1) [inline pkg/controller/translator/translator.go]: Instead of passing in all these Indexer's, we should just pass in the controller context and create the listers when we need them. So translator should be boiled down to just:

```
type Translator struct {
        namer utils.Namer
        ctx *context.ControllerContext
}

File: pkg/utils/serviceport.go [hunk 2]
```diff
@@ -39,6 +40,7 @@ type ServicePort struct {
 	Protocol      annotations.AppProtocol
 	SvcTargetPort string
 	NEGEnabled    bool
+	BackendConfig *backendconfigv1beta1.BackendConfig
```
[the remark this answers is not in the document]
Comment (MrHohn) [reply pkg/utils/serviceport.go]: One idea to make this a pointer is that we can distinguish if a ServicePort has any backendConfig associated with it.

File: pkg/controller/translator/translator.go [hunk 3]
```diff
@@ -34,43 +34,34 @@ import (
...
 type Translator struct {
-	namer          *utils.Namer
-	svcLister      cache.Indexer
-	nodeLister     cache.Indexer
-	podLister      cache.Indexer
-	endpointLister cache.Indexer
-	negEnabled     bool
+	namer      *utils.Namer
+	ctx        *context.ControllerContext
+	negEnabled bool
```
Comment (rramkumar1) [inline pkg/controller/translator/translator.go]: You can do this in a separate PR, but we can also put a flag that indicates whether NEG is enabled in the ContollerContext.

File: pkg/utils/serviceport.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (rramkumar1) [reply pkg/utils/serviceport.go]: Okay, we can leave it for now, but we should evaluate whether we really need it. I'd like to avoid having a pointer unless we actually have to modify the object.

[the remark this answers is not in the document]
Comment (MrHohn) [reply pkg/utils/serviceport.go]: That sounds good, thanks for the quick review!
